### PR TITLE
Refactor inbox

### PR DESCRIPTION
**Pull request by yenda**
Update inbox to use mailserver.request signals to increase reliability and responsiveness for mailserver requests (we now know when they fail and we know when they succeed).

Refactoring also includes:
- moving transport events to events.cljs
- removing message.v1 directory
- removing sym-key for 1-1 messages
- put filters in transport/filters
- add transport.inbox/topics to db to have individual counters per topic for inboxing

# Steps to test
- test mailbox in various conditions (app comes back from background, app restarted), no messages should be missing.
- the main visible change is that the `fetching messages` popup should disappear faster in most cases

status: ready

### CODE REVIEW

**Review by janherich — APPROVED** [review]

File: src/status_im/data_store/realm/schemas/account/migrations.cljs [hunk 1]
```diff
@@ -96,4 +96,7 @@
                   (aset message "content-type" "command")))))
 
 (defn v15 [old-realm new-realm]
-  (log/debug "migrating v13 account database"))
+  (log/debug "migrating v15 account database"))
+
+(defn v16 [old-realm new-realm]
+  (log/debug "migrating v15 account database"))
```
Comment (cammellos) [inline src/status_im/data_store/realm/schemas/account/migrations.cljs]: log should read v16

File: src/status_im/utils/utils.cljs [hunk 2]
```diff
@@ -66,6 +66,7 @@
  :utils/dispatch-later
  (fn [params]
    (doseq [{:keys [ms dispatch]} params]
+     (println :dispatch-later dispatch)
```
Comment (cammellos) [inline src/status_im/utils/utils.cljs]: debug statement

**Review by cammellos — APPROVED** [review]

**Comment by lukaszfryc** [comment]
@yenda does this PR fixes #1233 as well? I'm getting lot of `mailserver.request.expired` events on this PR build as well:

```
10-10 13:19:16.784 15226-15311/? D/ReactNativeJS: DEBUG [status-im.events:601] - :event-str {"type":"mailserver.request.expired","event":{"hash":"0xe1b1ec5d2fac17ea44650a88e1b2700b30e2190a2f607fd47a1ad16f737ff569"}}
10-10 13:19:16.786 15226-15311/? I/ReactNativeJS: INFO [status-im.transport.inbox:303] - offline inbox: message request expired for inbox topic 0xf676a41f from 1539083946 to 1539170346
10-10 13:19:16.790 15226-15311/? I/ReactNativeJS: INFO [status-im.transport.inbox:165] - offline inbox: request-messages for:   topic  0xf676a41f  from  1539083956  to    1539170356
10-10 13:19:16.794 15226-15311/? D/ReactNativeJS: DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :inbox.callback/request-messages-success
10-10 13:19:16.796 15226-15311/? I/ReactNativeJS: INFO [status-im.transport.inbox:310] - offline inbox: message request sent for inbox topic 0x2fdcb79d from 1539083956 to 1539170356
10-10 13:19:16.800 15226-15311/? D/ReactNativeJS: DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :signals/signal-received {"type":"mailserver.request.expired","event":{"hash":"0x8ba16db6a62306396dba4af82c940e442ffb472db80e27b7fffa90c4f46d8265"}}
10-10 13:19:16.802 15226-15311/? D/ReactNativeJS: DEBUG [status-im.events:601] - :event-str {"type":"mailserver.request.expired","event":{"hash":"0x8ba16db6a62306396dba4af82c940e442ffb472db80e27b7fffa90c4f46d8265"}}
10-10 13:19:16.805 15226-15311/? I/ReactNativeJS: INFO [status-im.transport.inbox:303] - offline inbox: message request expired for inbox topic 0xc9b6dc1a from 1539083946 to 1539170346
10-10 13:19:16.809 15226-15311/? I/ReactNativeJS: INFO [status-im.transport.inbox:165] - offline inbox: request-messages for:   topic  0xc9b6dc1a  from  1539083956  to    1539170356
10-10 13:19:16.812 15226-15311/? D/ReactNativeJS: DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :inbox.callback/request-messages-success
10-10 13:19:16.815 15226-15311/? I/ReactNativeJS: INFO [status-im.transport.inbox:310] - offline inbox: message request sent for inbox topic 0xd9b607e1 from 1539083956 to 1539170356
10-10 13:19:16.818 15226-15311/? D/ReactNativeJS: DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :signals/signal-received {"type":"mailserver.request.expired","event":{"hash":"0xae071c8640986c750285f4c687a4b9dab64ba43db144826de7d47375ed36412c"}}
10-10 13:19:16.821 15226-15311/? D/ReactNativeJS: DEBUG [status-im.events:601] - :event-str {"type":"mailserver.request.expired","event":{"hash":"0xae071c8640986c750285f4c687a4b9dab64ba43db144826de7d47375ed36412c"}}
10-10 13:19:16.824 15226-15311/? I/ReactNativeJS: INFO [status-im.transport.inbox:303] - offline inbox: message request expired for inbox topic 0x36211c37 from 1539083946 to 1539170346
10-10 13:19:16.828 15226-15311/? I/ReactNativeJS: INFO [status-im.transport.inbox:165] - offline inbox: request-messages for:   topic  0x36211c37  from  1539083956  to    1539170356
10-10 13:19:16.831 15226-15311/? D/ReactNativeJS: DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :inbox.callback/request-messages-success
10-10 13:19:16.834 15226-15311/? I/ReactNativeJS: INFO [status-im.transport.inbox:310] - offline inbox: message request sent for inbox topic 0x104c0f87 from 1539083956 to 1539170356
10-10 13:19:16.839 15226-15311/? D/ReactNativeJS: DEBUG [status-im.utils.handlers:37] - Handling re-frame event:  :signals/signal-received {"type":"mailserver.request.expired","event":{"hash":"0x59f97ad2d272cebd1d8723297a02084bb6069482e3fef2f5ac45bd39d975791f"}}
``` 

Here is the [TF session](https://app.testfairy.com/projects/4803590-status/builds/8622646/sessions/4403633331/?accessToken=1AMs0YopPqJSEH2b4Iwp8Ug48dA) when I installed the app, created new account and quickly joined couple of public channels. "Fetching messages" was shown for more than 3 mintues. But, all messages were fetched correctly as far as I can tell. Logs have `mailserver.request.expired`.

**Comment by lukaszfryc** [comment]
@yenda I also noticed 1 case when a message is marked as sent but it's not delivered (and probably not sent either). When app is connecting to peers, send the message and put the app into background for 1 minute. When you return, the message is marked as sent but never delivered.

---

Apart from that, all looks good. Here is full list of scenarios I tested:
- send msgs while connecting to peers
- send msgs while fetching messages
- kill the app, autologin and receive all messages
- kill the app, autologin and immediately send messages
- airplane mode, send messages, go online
- go offline then online and send messages
- receive PN while being in background
- go from bg and send messages
- retry sending message that was not sent 
- switch between mobile and wifi connection while sending and receiving msgs

**Comment by yenda** [comment]
@lukaszfryc is the message mark as sent an issue specific to that PR or does it also happen in develop ?

**Comment by lukaszfryc** [comment]
I think it maybe the case. Let me check

**Comment by lukaszfryc** [comment]
@yenda yes, it's on develop branch as well (I checked latest nightly). Do you plan to fix it here?

Btw, how about https://github.com/status-im/status-react/pull/6223#issuecomment-428537148?

**Comment by yenda** [comment]
@lukaszfryc no I won't fix it here there is already too many changes but I am looking at your first comment. It is normal that there is some requests that fails but it shouldn't fail for 3 minutes.

**Comment by yenda** [comment]
@lukaszfryc can you give it another try ?

**Comment by churik** [comment]
https://github.com/status-im/status-react/issues/6234 can be closed after merge, because of

>@churik I just spoke with @yenda and very soon we will fetch last 7 days worth of messages instead of 24h, which will mostly solve the original issue, as almost nobody will reply to message more then 7 days old.

**Comment by lukaszfryc** [comment]
@yenda there is still a lot of `mailserver.request.expired` events in [the log](https://app.testfairy.com/projects/4803622-status/builds/8624176/sessions/4403707459/?accessToken=-fzNKqTLj/lVIoTl75MondzcMGQ). The last commit did not help.

**Comment by lukaszfryc** [comment]
Btw, here is a separate issue related to messages not being sent after going from offline mode (https://github.com/status-im/status-react/pull/6223#issuecomment-428545755):  https://github.com/status-im/status-react/issues/6080

This can be fixed in a spearte PR.

**Comment by lukaszfryc** [comment]
> #6234 can be closed after merge, because of
> 
> > @churik I just spoke with @yenda and very soon we will fetch last 7 days worth of messages instead of 24h, which will mostly solve the original issue, as almost nobody will reply to message more then 7 days old.

@churik so this PR about inbox refactoring should fix the issue? Or maybe https://github.com/status-im/status-react/pull/6226 fixes it?

**Comment by churik** [comment]
@lukaszfryc this issue #6234  will be hardly reproducible when we will fetch 7d history instead of 24h.
It will be still possible but very unlikely that somebody will reply to massage older than 7 days.
And no, #6226 doesn't close it in full; only scenario 2.

**Comment by yenda** [comment]
@churik unfortunately this PR is not  fetching 7 days yet, it's just about using the signals to increase reliablity and efficiency of the mailbox. There is still a fix needed in status-go to allow to fetch more than 7 days and mailservers need to be updated after that

**Comment by churik** [comment]
Ok, so I'll be bad informed in #6223 
Thanks @yenda

**Comment by lukaszfryc** [comment]
@yenda there was Error encountered performing task 'cljsbuild' with profile(s): 'prod'

**Comment by lukaszfryc** [comment]
@yenda when I open lot of public chats, it sometimes takes 3-5 minutes to fetch messages. See https://app.testfairy.com/projects/4803590-status/builds/8624524/sessions/4403728713/?accessToken=hvs2LYCWgH0SBCWiwUoldd-VAqA

<blockquote><div><strong><a href="https://app.testfairy.com/projects/4803590-status/builds/8624524/sessions/4403728713/?accessToken=hvs2LYCWgH0SBCWiwUoldd-VAqA">TestFairy: Status 0.9.29 (10267) @2018</a></strong></div></blockquote>

**Pull request merged**